### PR TITLE
Add `CITATION.cff`

### LINKED ISSUE #86: Label all TODOs
There are several `TODO` comments in the code base. Each of these should refer to an issue.

### LINKED ISSUE #126: Missing mesh plot in ping pong demo
@jwallwork23 sorry, I just realised that you aren't showing the first two figures (the two meshes and source function). Is that intentional?

_Originally posted by @ddundo in https://github.com/mesh-adaptation/animate/issues/125#issuecomment-2114088081_

### LINKED ISSUE #127: Add `CITATION.cff` file
See discussions on https://github.com/mesh-adaptation/movement/issues/90 and https://github.com/mesh-adaptation/goalie/issues/115.

### PR DESCRIPTION
**Pull request by jwallwork23**
Closes #127.
Closes #86.
Closes #126.

Adds a citation file so that releases work nicely with Zenodo.

Reviewers: please check you are happy with how your names are presented, that your ORCIDs are correct and present (if you have one). Feel free to push directly to the branch if you have edits.

### CODE REVIEW

**Comment by jwallwork23** [comment]
Also pushed minor changes to label all TODOs with issues. Ready for review.

**Comment by jwallwork23** [comment]
Also fix for #126.

**Comment by ddundo** [comment]
Thanks @jwallwork23! I added my ORCID and pushed it here since you said we could :)

I also thought I'd add the other missing plot in the ping-pong demo (ping_pong-source_function.jpg) since it's a minor change, but then saw that in some parts of the demo you refer to it as a source function and in other as a sensor function. So the title of these plots is different in ping_pong-source_function.jpg and in ping_pong-final.jpg, even though they're the same function. Maybe choose one for consistency?

**Comment by ddundo** [comment]
Also, initially I didn't have `cffconvert` installed so I got this error:
```
Checking CITATION.cff formatting...
make: cffconvert: No such file or directory
make: *** [Makefile:32: check_citation] Error 127
```
but this message got lost in all the other messages that `make test` prints out.

Can we make it so that `make test` breaks if this an error occurs here, so that we can easily catch it? This is what happens if there is an error in the `lint` part of testing, for example.

**Comment by jwallwork23** [comment]
> I also thought I'd add the other missing plot in the ping-pong demo (ping_pong-source_function.jpg) since it's a minor change, but then saw that in some parts of the demo you refer to it as a source function and in other as a sensor function. So the title of these plots is different in ping_pong-source_function.jpg and in ping_pong-final.jpg, even though they're the same function. Maybe choose one for consistency?

@ddundo Thanks, fixed by 6aedaf3885fec6f08d2a7bfeb455398cd7418451.

**Comment by jwallwork23** [comment]
@acse-ej321 do you have an ORCID to add or can we merge this?

File: demos/ping_pong.py [hunk 1]
```diff
@@ -30,42 +30,49 @@
...
 #
-# Let's plot the sensor function as represented in :math:`\mathbb{P}1` spaces on mesh A.
-# ::
+# Let's plot it as represented in :math:`\mathbb{P}1` spaces on mesh A. ::
 
 V_A = FunctionSpace(mesh_A, "CG", 1)
 V_B = FunctionSpace(mesh_B, "CG", 1)
 x, y = SpatialCoordinate(mesh_A)
-sensor = Function(V_A, name="Sensor").interpolate(sin(pi * x) * sin(pi * y))
+source = Function(V_A, name="Sensor").interpolate(sin(pi * x) * sin(pi * y))
```
Comment (ddundo) [inline demos/ping_pong.py:47]:
```suggestion
source = Function(V_A, name="Source").interpolate(sin(pi * x) * sin(pi * y))
```

**Review by ddundo — APPROVED** [review]
Thanks @jwallwork23! I just pointed out one line where it still says "Sensor" :)

**Review by acse-ej321 — APPROVED** [review]

**Pull request merged**